app/services/integrations/aggregator/base_service.rb

```diff
@@ -50,6 +50,26 @@ def provider_key
         end
       end
 
+      def throttle!(*providers)
+        providers.each do |provider_name|
+          if provider == provider_name.to_s
+            raise BaseService::ThrottlingError unless Throttling.for(provider_name.to_sym).check(:client, throttle_key)
+          end
+        end
+      end
+
+      def throttle_key
+        # Hubspot and Xero calls are throttled globally, others are throttled per api key or token
+        case provider
+        when 'netsuite'
+          integration.token_id
+        when 'anrok'
+          integration.api_key
```

Comment:
This could potentially store api_keys in Redis, maybe that's a security issue? 🤔

Reply:
Good point!